src/skeletons.tsx

```diff
@@ -0,0 +1,184 @@
+
+    return (
+        <div
+            className={`${animationClasses.animation} ${styleClass.wrap}`}
+            role="status"
+            aria-hidden={ariaLabel === undefined}
+            aria-busy={ariaLabel !== undefined}
+            aria-label={ariaLabel}
+        >
+            <SkeletonBase></SkeletonBase>
+        </div>
+    );
+};
+
+type SkeletonTextProps = Omit<SkeletonLineProps, 'width'>;
+
+const SkeletonText = ({disableAnimation = false, ariaLabel}: SkeletonTextProps): JSX.Element => {
+    const animationClasses = useAnimation({disableAnimation});
+    const styleClass = useSkeletonStyles({height: 'fit-content'});
+
+    return (
+        <Stack
+            className={`${animationClasses.animation} ${styleClass.wrap}`}
+            space={16}
+            role="status"
+            aria-hidden={ariaLabel === undefined}
+            aria-busy={ariaLabel !== undefined}
+            aria-label={ariaLabel}
+        >
+            <SkeletonBase />
+            <SkeletonBase />
+            <SkeletonBase width="75%" />
+        </Stack>
+    );
+};
+
+type SkeletonCircleProps = Omit<SkeletonLineProps, 'width'> & {size?: number | string};
+
+const SkeletonCircle = ({
+    disableAnimation = false,
+    ariaLabel,
+    size = 40,
+}: SkeletonCircleProps): JSX.Element => {
+    const animationClasses = useAnimation({disableAnimation});
+    const styleClass = useSkeletonStyles({width: size, height: size});
+
+    return (
+        <div
+            className={classNames(animationClasses.animation, styleClass.wrap)}
+            role="status"
+            aria-hidden={ariaLabel === undefined}
+            aria-busy={ariaLabel !== undefined}
+            aria-label={ariaLabel}
+        >
+            <SkeletonBase height={size} width={size} radius="50%" />
+        </div>
+    );
+};
+
+const useSkeletonRowStyles = createUseStyles(() => ({
+    row: {
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+    circle: {
+        flexShrink: 0,
+        flexGrow: 0,
+    },
+    line: {
+        flexShrink: 1,
+        marginLeft: 16,
+    },
+}));
+
+type SkeletonRowProps = SkeletonLineProps;
+
+const SkeletonRow = ({
+    width = '100%',
+    disableAnimation = false,
+    ariaLabel,
+}: SkeletonRowProps): JSX.Element => {
+    const animationClasses = useAnimation({disableAnimation});
+    const styleClass = useSkeletonStyles({width, height: 'content-initial'});
+    const rowClass = useSkeletonRowStyles();
+
+    return (
+        <div
+            className={classNames(animationClasses.animation, styleClass.wrap, rowClass.row)}
+            role="status"
+            aria-hidden={ariaLabel === undefined}
+            aria-busy={ariaLabel !== undefined}
+            aria-label={ariaLabel}
+        >
+            <SkeletonBase height={40} width={40} radius="50%" className={rowClass.circle} />
+            <SkeletonBase width={width} className={rowClass.line} />
```

Comment:
I'd expect this to be a SkeletonCircle + SkeletonLine

Reply:
that was my very first thought, but then by reiterating I figured out that this approach would be better:

- aria attributes are only defined on the final element, the one that would wrap all the graphical components, whereas SkeletonBase does not have them
- the same for the animation: this way I make sure that all the graphical components that make up a skeleton will be flashing at the same pace. 

let's say I'm seeing the skeletons as a final class, wheresas a BaseSkeleton as an abstract one

Reply:
I see no problem in composing existing skeletons. If the aria-label is not given, the aria-hidden for inner elements would be "true"

you could also pass a role=presentation for child skeletons

Reply:
this means that if we don't pass a label a <SkeletonLine> would be aria-hidden in place of having a semantic element that tells the user "you should wait while we're charging this". I'd avoid this